src/net/udp.rs

```diff
@@ -85,6 +102,38 @@ mod test {
     }
 
     #[test]
+    fn send_receive_fragment_packet() {
+        let mut send_socket = UdpSocket::bind("127.0.0.1:12347").unwrap();
+        let mut recv_socket = UdpSocket::bind("127.0.0.1:12348").unwrap();
+
+        let addr = SocketAddr::new(
+            IpAddr::from_str("127.0.0.1").expect("Unreadable input IP."),
+            12348,
+        );
+
+        let handle = thread::spawn(move || {
+            loop {
+                let packet: io::Result<Option<Packet>> = recv_socket.recv();
+
+                match packet {
+                    Ok(Some(packet)) => {
+                        assert_eq!(packet.addr().to_string(), "127.0.0.1:12347");
+                        assert_eq!(packet.payload(), vec![123; 4000].as_slice());
+                        break;
+                    }
+                    Err(e) => { panic!(); },
```

Comment:
Why not just `unwrap` here?

Reply:
Since this is a test function that will gather bytes until it has builded up the packet. Then it will be Ok(Some()) than I check the payload from the packet